src/backend/executor/nodeShareInputScan.c

```diff
@@ -660,12 +657,11 @@ ExecSquelchShareInputScan(ShareInputScanState *node)
 		{
 			/* We are a consumer. Let the producer know that we're done. */
 			Assert(!local_state->closed);
-
 			local_state->ndone++;
 
 			if (local_state->ndone == local_state->nsharers)
 			{
-				shareinput_reader_notifydone(node->ref, sisc->nconsumers);
+				_shareinput_reader_notifydone(node->ref, sisc->nconsumers);
```

Comment:
Maybe we can use the new `shareinput_reader_notifydone` here? `local_state->closed` is false here, so the effect would be the same.